### PR TITLE
Implement caching/promisified API "get" method. Closes #15, #44.

### LINKED ISSUE #15: create ajax abstraction w/ caching

### LINKED ISSUE #44: Handle case where an API request returns a 401 due to revoked token
What should the behavior be?

Ref. https://github.com/bocoup/service-catalog/pull/41#discussion_r51741481

### PR DESCRIPTION
**Pull request by cowboy**
Also:

* Further generalize / simplify / awesomeify auth.
* Improve webpack debugging in development.
* Refactor components, simplifying where possible.

Closes #15.
Closes #44.

### CODE REVIEW

File: src/service/api.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Promise from 'bluebird';
+import superagent from 'superagent';
+import superagentCache from 'superagent-cache';
+import superagentPromise from 'superagent-promise';
+import superagentJsonapify from 'superagent-jsonapify';
+import auth from './auth';
+import {API_BASE_URL} from '../config.js';
+
+superagentJsonapify(superagent);
+superagentCache(superagent);
```
Comment (tkellen) [inline src/service/api.js]: what effect does this have?

**Comment by jennschiffer** [comment]
:+1: v nice

File: src/service/api.js (hunk 1 above)
Comment (cowboy) [inline src/service/api.js]: Caches responses in-memory https://github.com/jpodwys/superagent-cache#how-does-it-work

Comment (tkellen) [inline src/service/api.js]: :+1:

**Pull request merged**